freqtrade/freqtradebot.py

```diff
@@ -1736,7 +1736,7 @@ def fee_detection_from_trades(self, trade: Trade, order: Dict, order_amount: flo
                 trade.update_fee(fee_cost, fee_currency, fee_rate, order.get('side', ''))
 
         if not isclose(amount, order_amount, abs_tol=constants.MATH_CLOSE_PREC):
-            # TODO-lev: leverage?
+            # * Leverage could be a cause for this warning, leverage hasn't been thoroughly tested
```

Comment:
from discord

sam -  @richardjozsa You've been running freqtrade in futures mode right? Have you been getting any of this warning in your log output "Amount {amount} does not match amount {trade.amount}"
richardjozsa -  of course I'm using every time the same setup from config and strategy side, that worked earlier, when I used my crude solutions,margin_mode: "isolated", trading_mode: "futures.". I haven't got that message, that message dissapared after you repaired it.

Reply:
this is the fallback to fee fetching - and will only apply if  the order did not include fees.
I don't think that's the case on binance (although it may depend if it filled immediately).

So this is already a pretty rare case ...

Reply:
Ok, we should probably just leave the comment there, and once this is merged into stable it probably wouldnt be necessary anymore, this is just a heads up in case we arrive here